batch/python/tiles_geojson.py

```diff
@@ -0,0 +1,96 @@
+import json
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from typing import Any, Dict, List, Tuple
+
+from geojson import Feature, FeatureCollection
+from pyproj import CRS, Transformer
+from shapely.geometry import Polygon
+from shapely.ops import unary_union
+
+from errors import GDALError
+from gdal_utils import run_gdal_subcommand
+
+
+def to_4326(crs: CRS, x: float, y: float) -> Tuple[float, float]:
+    transformer = Transformer.from_crs(
+        crs, CRS.from_epsg(4326), always_xy=True
+    )
+    return transformer.transform(x, y)
+
+
+def extract_metadata_from_gdalinfo(gdalinfo_json: Dict[str, Any]) -> Dict[str, Any]:
+    """Extract necessary metadata from the gdalinfo JSON output."""
+    corner_coordinates = gdalinfo_json["cornerCoordinates"]
+
+    crs: CRS = CRS.from_string(gdalinfo_json["coordinateSystem"]["wkt"])
+    metadata = {
+        # NOTE: pixetl seems to always write features in tiles.geojson in
+        # epsg:4326 coordinates (even when the tiles themselves are
+        # epsg:3857). Reproduce that behavior for compatibility. If that
+        # ever changes, remove the call to to_4326 here.
+        "extent": [
+            *to_4326(crs, *corner_coordinates["lowerLeft"]),
+            *to_4326(crs, *corner_coordinates["upperRight"]),
+        ],
+        "name": gdalinfo_json["description"],
+    }
+
+    return metadata
+
+
+def process_file(file_path: str) -> Dict[str, Any]:
+    """Run gdalinfo and extract metadata for a single file."""
+    print(f"Running gdalinfo on {file_path}")
+    try:
+        stdout,stderr = run_gdal_subcommand(
```

Comment:
My mind went immediately to gdalinfo because (though it may do it differently in other places) pixetl gets metadata with gdalinfo here: https://github.com/wri/gfw_pixetl/blob/master/gfw_pixetl/utils/gdal.py#L170
I can verify that the results are the same with a few tiles.

Reply:
OK, thanks for that pointer to the pixetl code that is getting the meta-data.  I don't see any reprojection/resizing (to EPSG:4326 or meters) in that code you are pointing to, so does that mean you don't really need the to_4326 stuff?   See https://github.com/wri/gfw_pixetl/blob/cfb7c2fa1c0b6f366982e38839baa1cabc425938/gfw_pixetl/utils/gdal.py#L192  (But maybe I am missing a conversion somewhere else.)